### DISCUSSION TITLE
exception for 400 status code error

### DISCUSSION THREAD

**Discussion by ruhex** [datalust/seq-tickets#1491] [Help]
Hi, I'm using exception for logging 400 and 500 error in my .net core api. Byt seq write 400 error with 500 status code. Will I can chenge 'InternalError' type to 'BadRequest' for exception ?

**Comment by nblumhardt** [comment]
Hi Mikhail,

It sounds like you might be using the _Serilog.AspNetCore_ exception handling middleware - is that correct?

If so, you may need to move it up to the start of your middleware pipeline, so that the exception is handled and converted into a 400 response before it reaches the middleware. You'll also need, in that case, to separately log the exception with an error when it is converted to a status code (the middleware won't then see the exception object).

Another option is to use the sample middleware code from https://github.com/datalust/serilog-middleware-example directly, and customize it to recognize your exception types.

Hope this helps,
Nick
